meta-lmp-bsp/recipes-bsp/u-boot/u-boot-ostree-scr-fit/imx8mmevk/boot.cmd

```diff
@@ -5,17 +5,96 @@ setenv bootlimit 3
 setenv devtype mmc
 setenv devnum 2
 
+# uEnv handing
 setenv bootcmd_resetvars 'setenv kernel_image; setenv bootargs; setenv kernel_image2; setenv bootargs2'
 setenv bootcmd_otenv 'run bootcmd_resetvars; ext4load ${devtype} ${devnum}:2 ${loadaddr} /boot/loader/uEnv.txt; env import -t ${loadaddr} ${filesize}'
+setenv bootcmd_getroot 'setexpr ostree_root gsub "^.*ostree=([^ ]*).*$" "\\\\1" "${bootargs}"'
+
+# U-Boot image update helpers
+setenv uboot_hwpart 1
+setenv load_image 'ext4load ${devtype} ${devnum}:2 ${loadaddr} ${image_path}'
+setenv set_blkcnt 'setexpr blkcnt ${filesize} + 0x1ff && setexpr blkcnt ${blkcnt} / 0x200'
+setenv update_image 'run set_blkcnt && mmc dev ${devnum} ${uboot_hwpart} && mmc write ${loadaddr} ${start_blk} ${blkcnt}'
+
+# FIO bootcount, upgrade_available and rollback initialization
+mmc rescan
+fiovb init {devnum}
+fiovb is_secondary_boot
+
+run bootcmd_otenv
+run bootcmd_getroot
+
+if test ${fiovb.is_secondary_boot} -eq 0; then
+	fiovb read_pvalue bootcount 4
+	if test ! $? -eq 0; then fiovb write_pvalue bootcount 0; fi
```

Comment:
what would happen if any of the fiovb calls fail? how is the error handling implemented

Reply:
good point

to be honest I've borrowed that from existing boot.cmd script for Schneider Hub.
`$?` is a result of previous command, we can also check that result for `fiovb write_pvalue bootcount 0` and  if it fails just stop running script and report critical error

Reply:
since now we are implementing some sort of state machine just making sure we are not entering some zombie state; can I assume this is under control so we dont need to worry about it?